components/sync15/src/sync_multiple.rs

```diff
@@ -110,6 +120,19 @@ fn do_sync_multiple(
         sync_result.service_status = ServiceStatus::Interrupted;
         return Ok(());
     }
+
```

Comment:
IIUC, this `do_sync_multiple` is receiving a `Sync15StorageClientInit` instance as an argument, and this instance contains `key_id` and `access_token` fields, could we tie it back to one of those fields as well for completeness? I don't understand the code very well here, but it seems like the "if our storage_init has changed we can't reuse the client" logic below was valid, but it should also have been clearing the in-memory cached state.

Reply:
> but it seems like the "if our storage_init has changed we can't reuse the client" logic below was valid, but it should also have been clearing the in-memory cached state.

hrmph, yes, that's very true.